traffic_router/core/src/main/java/org/apache/traffic_control/traffic_router/core/dns/NameServer.java

```diff
@@ -328,12 +329,18 @@ private static RRset setNegativeTTL(final RRset original, final int flags) {
 			if (record instanceof SOARecord) {
 				final SOARecord soa = (SOARecord) record;
 
+				// Set the "minimum" attribute to be the maximum of the current minimum value and 900 (seconds)
+				// This is done to increase the negative caching TTL, so as to maximize the time interval between
+				// successive NXDOMAIN or NXRRSET responses.
+				final long minimum = Math.max(soa.getMinimum(), negativeCachingTTL);
+				long ttl = soa.getTTL();
 				// the value of the minimum field is less than the actual TTL; adjust
 				if (soa.getMinimum() != 0 || soa.getTTL() > soa.getMinimum()) {
-					record = new SOARecord(soa.getName(), DClass.IN, soa.getMinimum(), soa.getHost(), soa.getAdmin(),
-							soa.getSerial(), soa.getRefresh(), soa.getRetry(), soa.getExpire(),
-							soa.getMinimum());
-				} // else use the unmodified record
+					ttl = soa.getMinimum();
+				}
+				record = new SOARecord(soa.getName(), DClass.IN, ttl, soa.getHost(), soa.getAdmin(),
+						soa.getSerial(), soa.getRefresh(), soa.getRetry(), soa.getExpire(),
+						minimum);
```

Comment:
If `soa.getMinimum()` is greater than `negativeCachingTTL`, this will end up generating a record where the minimum field is less than the TTL. So `minimum` should be used in the condition, in the `ttl` assignment, and the new record instead of `soa.getMinimum()`, unless I'm missing something?